_posts/2017-03-23-first-ocim-blog.markdown

```diff
@@ -0,0 +1,107 @@
+Cypher has always been a language that operates on a single implicitly understood graph, and producing a stream or records as output (the collection of which effectively form a table).
+Alastair Green discussed motivations for changing this model to make Cypher a language _closed over graphs_, sketching out a future where Cypher would come to operate on a multitude of graphs, and also producing a multitude of graphs as output.
+Alastair touched on important subtopics such as identity, addressing, and how compositions of graphs from distinct sources may be defined.
+
+Stefan Plantikow led a longer session on the very topic, presenting his latest thinking on how Cypher can be remodeled towards a graph-in-graph-out paradigm.
+Stefan deep-dove into the motivations, but the focus was put on how the concept of multiple graphs could be made logically consistent, and how the execution model could be extended, while still paying respects to Cypher's sizable user base and imposing breaking change in semantics.
+One of Stefan's main ideas was the re-interpretation of Cypher's result records as graph-records, or 'g-records' (should we retro-actively amend this to mention our new term graphlets?), meaning each binding of a matched subgraph would itself be interpreted as a (typically very small) graph, and the extended ability to collapse/union all such g-records into one large result graph.
+Both the g-record model and it's larger sibling would enact Cypher as being _closed over graphs_, as it would be possible to upon retrieval of the result graph(s) immediately issue a new Cypher query, now pattern matching on the newly computed results.
+Stefan also gave detailed syntax proposals for how to define and use graphs as values in the context of a query, including a take on the addressing topic raised by Alastair.
+It was made clear that this topic is at the top of the minds of several key Cypher innovators.
+
+Before Stefan's dive into multi-graph land, Martin Junghanns from the University of Leipzig presented his research project on implementing Cypher on Gradoop, a graph platform based on Apache Hadoop.
+Martin gave us an overview on how their system handled query planning, and the model used to represent (intermediate) query results in the distributed framework, Flink, that queries are executed in.
+The project had also featured some interesting extensions to the Property Graph Model, over which Cypher is defined, including a concept of logical subgraphs and a set of graph operations.
+
+Following Martin's talk, Hannes Voigt from the Technical University of Dresden walked us through his research project, in which also Michael Hunger from Neo had participated.
+The topic was virtual graphs and views, and featured several intriguing extensions to Cypher, which were expressed as able to cross the concept chasm, which Hannes explained as the different levels of abstraction that users view their data in.
```

Comment:
"Cypher, which were expressed as able to cross the concept chasm"
->
"Cypher. These were expressed in terms of 'crossing the concept chasm' "